modules/cloud-dataplex/variables.tf

```diff
@@ -45,8 +53,10 @@ variable "zones" {
   type = map(object({
     type      = string
     discovery = optional(bool, true)
+    iam       = optional(map(list(string)), null)
     assets = map(object({
-      bucket_name            = string
+      resource_name          = string
+      resource_project       = optional(string)
       cron_schedule          = optional(string, "15 15 * * *")
       discovery_spec_enabled = optional(bool, true)
       resource_spec_type     = optional(string, "STORAGE_BUCKET")
```

Comment:
Shouldn't we add now conditional validation on types of resources? That only `STORAGE_BUCKET`  and `BIGQUERY_DATASET` are valid?